misc-utils/lsfd-sock-xinfo.c

```diff
@@ -1548,3 +1621,405 @@ static void load_xinfo_from_proc_netlink(ino_t netns_inode)
+#endif
+#ifdef ETH_P_IFE
+	case ETH_P_IFE:
+		return "ife";
+#endif
+	case ETH_P_AF_IUCV:
+		return "af_iucv";
+#ifdef ETH_P_802_3_MIN
+	case ETH_P_802_3_MIN:
+		return "802_3_min";
+#endif
+	default:
+		return "unknown";
+	}
+}
+
+static char *packet_get_name(struct sock_xinfo *sock_xinfo,
+			     struct sock *sock __attribute__((__unused__)))
+{
+	struct packet_xinfo *pkt = (struct packet_xinfo *)sock_xinfo;
+	char *str = NULL;
+	const char *type = sock_decode_type(pkt->type);
+	const char *proto = packet_decode_protocol(pkt->protocol);
+	const char *iface = get_iface_name(sock_xinfo->netns_inode,
+					   pkt->iface);
+
+	if (iface && proto)
+		xasprintf(&str, "type=%s protocol=%s iface=%s",
+			  type, proto, iface);
+	else if (proto)
+		xasprintf(&str, "type=%s protocol=%s",
+			  type, proto);
+	else if (iface)
+		xasprintf(&str, "type=%s iface=%s",
+			  type, iface);
+	else
+		xasprintf(&str, "type=%s", type);
+
+	return str;
+}
+
+static char *packet_get_type(struct sock_xinfo *sock_xinfo __attribute__((__unused__)),
+			     struct sock *sock __attribute__((__unused__)))
+{
+	const char *str;
+	struct packet_xinfo *pkt = (struct packet_xinfo *)sock_xinfo;
+
+	str = sock_decode_type(pkt->type);
+	return xstrdup(str);
+}
+
+static bool packet_fill_column(struct proc *proc __attribute__((__unused__)),
+			       struct sock_xinfo *sock_xinfo,
+			       struct sock *sock __attribute__((__unused__)),
+			       struct libscols_line *ln __attribute__((__unused__)),
+			       int column_id,
+			       size_t column_index __attribute__((__unused__)),
+			       char **str)
+{
+	struct packet_xinfo *pkt = (struct packet_xinfo *)sock_xinfo;
+
+	switch (column_id) {
+	case COL_PACKET_IFACE: {
+		const char *iface;
+		iface = get_iface_name(sock_xinfo->netns_inode,
+				       pkt->iface);
+		if (iface) {
+			*str = xstrdup(iface);
+			return true;
+		}
+		break;
+	}
+	case COL_PACKET_PROTOCOL: {
+		const char *proto;
+		proto = packet_decode_protocol(pkt->protocol);
+		if (proto) {
+			*str = xstrdup(proto);
+			return true;
+		}
+		break;
+	}
+	default:
+		break;
+	}
+	return false;
+}
+
+static const struct sock_xinfo_class packet_xinfo_class = {
+	.get_name = packet_get_name,
+	.get_type = packet_get_type,
+	.get_state = NULL,
+	.get_listening = NULL,
+	.fill_column = packet_fill_column,
+	.free = NULL,
+};
+
+static void load_xinfo_from_proc_packet(ino_t netns_inode)
+{
+	char line[BUFSIZ];
+	FILE *packet_fp;
+
+	packet_fp = fopen("/proc/net/packet", "r");
+	if (!packet_fp)
+		return;
+
+	if (fgets(line, sizeof(line), packet_fp) == NULL)
+		goto out;
+	if (!(line[0] == 's' && line[1] == 'k'))
+		/* Unexpected line */
+		goto out;
+
+	while (fgets(line, sizeof(line), packet_fp)) {
+		uint16_t type;
+		uint16_t protocol;
+		unsigned int iface;
+		unsigned long inode;
+		struct packet_xinfo *pkt;
+
+		if (sscanf(line, "%*x %*d %" SCNu16 " %" SCNx16 " %" SCNu32 "%*d %*d %*d %lu",
+			   &type, &protocol, &iface, &inode) < 4)
```

Comment:
iface is an unsigned int but uses the uint32 format

Reply:
Done. I used `%u` instead.